### ISSUE TITLE
On update, run into a hostpci error and cloud-init image cannot be replaced.

### ISSUE THREAD

**Issue by spott** [muhlba91/pulumi-proxmoxve#28]
After updating to 2.3.0 (from 2.0.0), I'm getting:

```
pulumi:pulumi:Stack (vpn-vpn-infra):
    QH5RPe2C2+yuHOLKkDfgButgt1fuF9wYZieuhAj4pyc=

    error: update failed

  proxmoxve:VM:VirtualMachine (vpn):
    error: 1 error occurred:
    	* creating urn:pulumi:vpn-infra::vpn::proxmoxve:VM/virtualMachine:VirtualMachine::vpn: 1 error occurred:
    	* received an HTTP 400 response - Reason: Parameter verification failed. (hostpci0: invalid format - missing key in comma-separated list property)
```

Not sure if there is a fix or not.

edit:

There appears to be a problem with replacing the cloud-init image as well.

**Comment by muhlba91** [comment]
I can verify this as well. Also the cloud-init image cannot be replaced.
The upstream Terraform provider seems to have changes behavior - with 2.2.0 it restarted the machines on a cloud-init change.

Can you please also raise this in the upstream provider and reference here?

**Comment by danielr1996** [comment]
I got the same error on 2.3.0, tried downgrading to 2.2.0 and the hostpci error disappeared again. 
However I still get the following error: 
```
 error: 1 error occurred:
        * creating urn:pulumi:dev::k8s::proxmoxve:VM/virtualMachine:VirtualMachine::k8s-dev-dataplane-0: 1 error occurred:
        * received an HTTP 500 response - Reason: lvcreate 'pve/vm-103-cloudinit' error:   Logical Volume "vm-103-cloudinit" already exists in volume group "pve"
```

Is that the cloudinit error you mentioned? 

The strange thing is that I also tried it with an older project that I had created with 2.2.0, but abandoned again a few weeks ago and get the same cloudinit error. So I'm wondering if thats the problem with the upstream provider or something went wrong on my side. I can confirm that the volume isn't there (at least not listed in the proxmox storage view) and then gets created by pulumi. 

@spott would you mind updating the title and description to include the cloudinit error, I think this would helps others searching the issues for that particular error.

**Comment by muhlba91** [comment]
yes, that‘s the cloud-init error. in my tests it worked but maybe it was also due to a specific constellation of PVE version, upstream provider, and VM states.

the cloud-init though gets changed and after a stop and start of the VM these values get set correctly.

**Comment by qazbnm456** [comment]
Per the problem you mentioned initially @spott, I sent the PR to fix it upstream and got it merged already.

@muhlba91 The upstream yesterday released a newer version that includes the fix. Could you kindly update the dependency? Thanks!

**Comment by spott** [comment]
@qazbnm456 Thanks!

**Comment by muhlba91** [comment]
thanks for the info! update will come friday or saturday. (i‘m currently not home.)

**Comment by muhlba91** [comment]
`v2.3.1` is released which used `0.12.1` as upstream version.
i can confirm the `hostpci` error did not appear for me anymore; however, the cloud-init one still persists.

**Comment by bpg** [comment]
Hey @muhlba91 / @danielr1996 👋🏼 
Would you mind to open an issue in https://github.com/bpg/terraform-provider-proxmox with more details about the "cloud-init" error? Ideally, an example of the template that was causing the error.
Thanks!
